### ISSUE TITLE
Covert from JAX Arrays!

### ISSUE THREAD

**Issue by clarkmiyamoto** [arviz-devs/arviz#2480]
## Tell us about it

Hi Arviz Team / Community,

I was wondering if you folks were interested in adding native support for `jax.numpy.ndarray`s?

Currently I always have code like:
```
samples # Type (jax.numpy.ndarray)
samples = np.array(samples)
arviz.convert_to_dataset(samples)
```
But life could be much simpler
```
samples # Type (jax.numpy.ndarray)
arviz.convert_to_dataset(samples)
```
Curious on your folks thoughts! Let me know if this is useful, or I am at the pinnacle of lazyness.

**Comment by jordandeklerk** [comment]
Thanks for raising this @clarkmiyamoto! Tagging @OriolAbril for visibility/thoughts on this. 

Yes, I think this would be useful and aligns with how people use ArviZ.

I think we can just accept “array‑like” inputs and normalize them to NumPy at the conversion boundary. We can make `convert_to_datatree` accept objects that implement `__array__` (not just `np.ndarray`) and pass `np.asarray(obj)` through to our existing machinery. To make `dicts` of JAX arrays work consistently, we can also normalize inside `ndarray_to_dataarray` before building the `DataArray`. This doesn’t add a JAX dependency and matches what we already do in the NumPyro converter, where we bring device arrays to host first.

I think with that change, your example would work as written without the intermediate `np.array(...)` call. Longer‑term, keeping JAX arrays as duck arrays throughout ArviZ would need a broader audit, so this normalization feels like the safest quick win IMO.

---

Also, leaving a link to the migration guide for some major on-going changes within ArviZ that you may find useful! https://python.arviz.org/en/stable/user_guide/migration_guide.html

**Comment by OriolAbril** [comment]
I also think it would be nice to support something like this. Not completely sure about what attributes/methods to use for that. I am very confused with the current state of `__array__`, array api, presence of a `to_numpy()` method...

Longer term it would also be nice to allow using jax arrays and the underlying data structure within the xarray objects, we sent a proposal that included something like that: https://github.com/numfocus/small-development-grant-proposals/issues/39 (for now unfunded). It is also not completely clear how feasible that is right now.

**Comment by clarkmiyamoto** [comment]
So from my understanding, the proper way to do this is to use `__array___()`. For example
```
arr_jnp = jnp.array([1,2,3])
arr_np = arr_jnp.__array__()
type(arr_np) # Prints out `np.ndarray`
```
See https://github.com/jax-ml/jax/issues/16054 for discussion.

I like the solution that @jordandeklerk suggested, that is modify `convert_to_datatree ` to work w/ objects that have the method `__array__`. Here's some explicit code which would go in https://python.arviz.org/en/latest/_modules/arviz/data/converters.html#convert_to_inference_data
```
elif hasattr(obj, "__array__") and callable(getattr(obj, "__array__")):
    obj = obj.__array__()
    dataset = dict_to_dataset({"x": obj}, coords=coords, dims=dims)
```
How do y'all feel about this? If it's acceptable I'll PR

**Comment by jordandeklerk** [comment]
This sounds good to me. I think we can work out the details and any modifications in the PR.
